plugins/node/opentelemetry-instrumentation-pg/test/pg.test.ts

```diff
@@ -563,5 +563,113 @@ describe('pg', () => {
+      };
+      const events: TimedEvent[] = [];
+      const span = tracer.startSpan('test span');
+      await context.with(trace.setSpan(context.active(), span), async () => {
+        try {
+          const resPromise = await client.query(query);
+          assert.ok(resPromise);
+          runCallbackTest(span, attributes, events);
+        } catch (e) {
+          assert.ok(false, e.message);
+        }
+      });
+      assert.strictEqual(called, true);
+    });
+
+    it('should call postQueryHook with query text and params if set', async () => {
+      instrumentation.disable();
+      let called = false;
+      const values = ['0'];
+      const query = 'SELECT $1::text';
+      const config: PgInstrumentationConfig = {
+        postQueryHook: ctx => {
+          called = true;
+          assert.strictEqual(ctx.query, query);
+          assert.strictEqual(ctx.params, values);
+        },
+      };
+      instrumentation.setConfig(config);
+      instrumentation.enable();
+
+      const attributes = {
+        ...DEFAULT_ATTRIBUTES,
+        [SemanticAttributes.DB_STATEMENT]: query,
+      };
+      const events: TimedEvent[] = [];
+      const span = tracer.startSpan('test span');
+      await context.with(trace.setSpan(context.active(), span), async () => {
+        const resPromise = await client.query(query, values);
+        try {
+          assert.ok(resPromise);
+          runCallbackTest(span, attributes, events);
+        } catch (e) {
+          assert.ok(false, e.message);
+        }
+      });
+      assert.strictEqual(called, true);
+    });
+
+    it('should call postQueryHook with query config if set', async () => {
+      instrumentation.disable();
+      const name = 'fetch-text';
+      const query = 'SELECT $1::text';
+      const values = ['0'];
+      let called = false;
+      const config: PgInstrumentationConfig = {
+        postQueryHook: ctx => {
+          called = true;
+          if (!ctx.config) {
+            assert.ok(false, 'ctx.config was undefined');
+          }
+          assert.strictEqual(ctx.config.text, query);
+          assert.strictEqual(ctx.config.values, values);
+        },
+      };
+      instrumentation.setConfig(config);
+      instrumentation.enable();
+
+      const attributes = {
+        ...DEFAULT_ATTRIBUTES,
+        [AttributeNames.PG_PLAN]: name,
+        [SemanticAttributes.DB_STATEMENT]: query,
+      };
+      const events: TimedEvent[] = [];
+      const span = tracer.startSpan('test span');
+
+      await context.with(trace.setSpan(context.active(), span), async () => {
+        try {
+          const resPromise = await client.query({
+            name: name,
+            text: query,
+            values: values,
+          });
+          assert.strictEqual(resPromise.command, 'SELECT');
+          runCallbackTest(span, attributes, events);
+        } catch (e) {
+          assert.ok(false, e.message);
+        }
+      });
+      assert.strictEqual(called, true);
+    });
```

Comment:
I also suggest having some negative tests for the hook - what happens if it throws? what if it's not a function?
This introduced new logic is currently not covered with tests.
You can see an example of that on the [responseHook tests here](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/81b3190af64bda14f87c5b0cbd6172bafda26408/plugins/node/opentelemetry-instrumentation-pg/test/pg.test.ts#L461)